File: 프로그래머스/lv2/권채린/다음 큰 숫자.cpp
#include <string>
#include <vector>

using namespace std;

int binary(int num) {
    string bi = "";
    int count=0;
    
    while(num>0) {
        bi = to_string(num%2)+bi;
        if( num%2 == 1) {
            count++;
        }
        num /= 2;
    }
    return count;
    
}


int solution(int n) {
    int answer = n;
    int goal = binary(n);
    
    while(1) {
        answer++;
        if(goal == binary(answer))
            break;
    }
    
    return answer;
    
}

File: 프로그래머스/lv2/장정윤/다음 큰 숫자.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12911
def solution(n):
    answer = 0
    # 2진수로 변환, 문자열로 바꿔서 1 카운트
    binary = bin(n)
    cnt = n + 1
    while cnt < 1000001:
        if str(binary).count('1') == str(bin(cnt)).count('1'):
            return cnt
        else:    
            cnt += 1